test/tandem/testTandemSimulator.js

```diff
@@ -368,81 +367,117 @@ describe('tandemSimulator.js', function() {
-      var regBasal2 = builder.makeScheduledBasal()
-        .with_time('2014-09-25T18:40:00.000Z')
-        .with_deviceTime('2014-09-25T18:40:00')
+      var suppressed2 = builder.makeScheduledBasal()
+        .with_time('2014-09-25T18:30:00.000Z')
+        .with_deviceTime('2014-09-25T18:30:00')
         .with_timezoneOffset(0)
         .with_conversionOffset(0)
-        .with_rate(1.3)
-        .with_scheduleName('billy');
-      var thisSim = pwdSimulator.make({settings: settings});
-      var expectedFirstBasal = _.cloneDeep(regBasal1);
-      expectedFirstBasal = expectedFirstBasal.set('duration', 300000).done();
-      var expectedSecondBasal = _.cloneDeep(tempBasal);
-      expectedSecondBasal.set('previous', expectedFirstBasal);
-      expectedSecondBasal.suppressed = expectedSecondBasal.suppressed
-        .set('scheduleName', 'billy').done();
-      expectedSecondBasal = expectedSecondBasal.done();
-      var expectedThirdBasal = _.cloneDeep(regBasal2);
-      expectedThirdBasal = expectedThirdBasal.set('duration', 19200000)
-        .set('previous', _.omit(expectedSecondBasal, 'previous'))
-        .done();
-      expectedThirdBasal.annotations = [{code: 'insulet/basal/fabricated-from-schedule'}];
-      thisSim.basal(regBasal1);
-      thisSim.basal(tempBasal);
-      thisSim.basal(regBasal2);
-      thisSim.finalBasal();
-      expect(thisSim.getEvents()).deep.equals([
-        expectedFirstBasal,
-        expectedSecondBasal,
-        expectedThirdBasal
+        .with_rate(2)
+        .with_duration(1800000);
+      var tempBasal = builder.makeTempBasal()
+        .with_time('2014-09-25T18:15:00.000Z')
+        .with_deviceTime('2014-09-25T18:15:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0)
+        .with_duration(1800000)
+        .with_previous(suppressed1.done());
+      var tempBasalStart = {
+            type: 'temp-basal',
+            subType: 'start',
+            percent: 0.65,
+            duration: 1800000
+          };
+      var tempBasalStop = {
+            type: 'temp-basal',
+            subType: 'stop',
+            time_left: 0
+          };
+
+
+      var basal3 = builder.makeScheduledBasal()
+        .with_time('2014-09-25T19:40:00.000Z')
+        .with_deviceTime('2014-09-25T19:40:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0)
+        .with_rate(2)
+        .with_duration(1800000);
+
+
+      var expectedTempBasal = tempBasal.with_payload({duration:1800000})
+                                        .set('percent',0.65)
+                                        .done();
+
+      simulator.basal(suppressed1);
+      simulator.tempBasal(tempBasalStart);
+      simulator.basal(tempBasal);
+      simulator.basal(suppressed2);
+      simulator.tempBasal(tempBasalStop);
+      simulator.basal(basal3);
+      expect(simulator.getEvents()).deep.equals([
+        suppressed1.done(),
+        expectedTempBasal,
+        suppressed2.done()
```

Comment:
I'm having trouble with this one. I'm expecting to see another expected temp basal since it's crossing a schedule.